src/styles/main.scss

```diff
@@ -1,3 +1,71 @@
+$square-side: 100px;
+$calendar-padding: 10px;
+$calendar-gap: 1px;
+$calendar-width: ($square-side * 7)
+  + ($calendar-padding * 2) + ($calendar-gap * 6);
+$days-names: mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6;
+
 body {
   margin: 0;
+
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.wrapper {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  box-sizing: border-box;
+
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+
+  width: $calendar-width;
+  padding: $calendar-padding;
+
+  &__day {
+    box-sizing: border-box;
+    cursor: pointer;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    width: $square-side;
+    height: $square-side;
+    border: 1px solid #000;
+    background-color: #eee;
+
+    transition: 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      background-color: #ffc0cb;
+    }
+
+  }
+
+  @each $day, $number in $days-names {
+    &--start-day--#{$day} > :first-child {
+      margin-left: calc((#{$square-side + $calendar-gap}) * #{$number});
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length--#{$i} > :not(:nth-child(-n + #{$i})) {
+      visibility: hidden;
```

Comment:
`display: none` is a better choice here - please spend some time to find out why so.